R/load_files.R

```diff
@@ -1269,7 +1269,7 @@ check_metadata_headers <- function(corrected_names, df_name) {
     case_match <- setdiff(grep(grep_pattern, corrected_names, ignore.case = TRUE),
                           exact_match_grep)
     if (isTRUE(length(exact_match_grep) == 1 ||
-               corrected_names[case_match] %in% controlled_headers))
+               corrected_names[case_match] %in% unlist(controlled_headers)))
```

Comment:
Reason:
```
> ls_x <- list(A = "a",
+              B = c("b", "bb"),
+              C = "c")
> "c" %in% ls_x
[1] TRUE
> "b" %in% ls_x
[1] FALSE
> "b" %in% unlist(ls_x)
[1] TRUE
```

For checking "Template" `case_match` return value for `grep_pattern = "Barcode$|Plate$"` but `corrected_names[case_match] %in% controlled_headers))` returns FALSE. As result "Template" was changed into "Barcode"

Reply:
"Template was changed into Barcode"? 🤔